### DISCUSSION TITLE
What does the SD Card do?

### DISCUSSION THREAD

**Discussion by vaderag** [darthcloud/BlueRetro#39] [Q&A] [answered]
I've just been reading through the docs and trying to establish the purpose of connecting the SD Card as shown [here](https://github.com/darthcloud/BlueRetroHW/blob/master/DIY/BlueRetroDIY.pdf)

Reading through everything however I don't see anywhere it's used (it's definitely possible i'm missing something!)
Is it required?

**Comment by darthcloud** [comment, marked as answer]
The filesystem is currently used to save the configuration done via the web configuration interface.
In the future emulated memory card saves will go there to.

I make two type of build currently one with FS on sdcard and theother with FS on the ESP32 flash.

on ESP32 flash is convenient as you need nothing else than the ESP32.

SD is convenient since you can easily backup configuration.

In the future the FS will be exposed via the web config page so at that point I might drop the SD options. So far it's not a popular option ;)

**Reply by vaderag** [reply]
Ah okay, thanks.
So presumably if I plan to use it _only_ for BlueRetro then I can just **not** wire in the SD card adaptor and everything will work fine :)

Unrelated, but while your here (thank you very much btw!) is there a preferred board out of the three you list?

**Reply by darthcloud** [reply]
Yeah no need for SD with SPIFFS FW, get https://www.digikey.ca/en/products/detail/espressif-systems/ESP32-DEVKITC-32E/12091810

**Reply by vaderag** [reply]
> Yeah no need for SD with SPIFFS FW, get https://www.digikey.ca/en/products/detail/espressif-systems/ESP32-DEVKITC-32E/12091810

I noticed this:
SD card is required to be able to save the Bluetooth pairing keys required by some controllers for quick reconnect

Is there a list of controllers that's needed for? (Or ones that it's definitely _not_ needed for?)

**Reply by darthcloud** [reply]
This is just outdated wording before i added the internal flash option.

With spiffs FW everything is save on internal flash. With SD FW everything on SD. You don't need SD card.

Edit: I updated wiki to remove this deprecated warning.
